src/transformers/modeling_outputs.py

```diff
@@ -812,3 +812,36 @@ class Seq2SeqQuestionAnsweringModelOutput(ModelOutput):
     encoder_last_hidden_state: Optional[torch.FloatTensor] = None
     encoder_hidden_states: Optional[Tuple[torch.FloatTensor]] = None
     encoder_attentions: Optional[Tuple[torch.FloatTensor]] = None
+
+
+@dataclass
+class SemanticSegmentationModelOutput(ModelOutput):
+    """
+    Base class for outputs of instance segmentation models.
+
+    Args:
+        loss (`torch.FloatTensor` of shape `(1,)`, *optional*, returned when `labels` is provided):
+            Classification (or regression if config.num_labels==1) loss.
+        logits (`torch.FloatTensor` of shape `(batch_size, config.num_labels, height, width)`):
+            Classification scores for each pixel.
+        legacy_logits (`torch.FloatTensor` of shape `(batch_size, config.num_labels, height/4, width/4)`):
+            For backward compatibility, old logits returned by `BeitForSemanticSegmentation` and
+            `SegformerForSemanticSegmentation`.
```

Comment:
Not sure if we should keep this. We could also call it `raw_logits` (which are the raw logits coming out of the model without any interpolation).

Reply:
I have a doubt about the naming. `logits` is usually used to indicate un-normalized vectors used to compute a downstream task (e.g. classification after a softmax layer). If that is the case, return this output class is unfeasible for some models, e.g. `MaskFormer`. 

In `MaskFormer` we compute the final segmentation masks using two vectors

```python
def forward(self, *args, **kwargs):
        outputs: MaskFormerOutput = self.model(*args, **kwargs)
        # mask classes has shape [BATCH, QUERIES, CLASSES + 1]
        # remove the null class `[..., :-1]`
        masks_classes: Tensor = outputs.preds_logits.softmax(dim=-1)[..., :-1]
        # mask probs has shape [BATCH, QUERIES, HEIGHT, WIDTH]
        masks_probs: Tensor = outputs.preds_masks.sigmoid()
        # now we want to sum over the queries,
        # $ out_{c,h,w} =  \sum_q p_{q,c} * m_{q,h,w} $
        # where $ softmax(p) \in R^{q, c} $ is the mask classes
        # and $ sigmoid(m) \in R^{q, h, w}$ is the mask probabilities
        # b(atch)q(uery)c(lasses), b(atch)q(uery)h(eight)w(idth)
        segmentation: Tensor = torch.einsum("bqc, bqhw -> bchw", masks_classes, masks_probs)

        return MaskFormerForSemanticSegmentationOutput(segmentation=segmentation, **outputs)
```        
Returning a vector `logits` of shape `batch_size, config.num_labels, height, width)` is impossible due. I see two possible solutions

Reply:
We might have to treat `MaskFormer` with an exception (in the sense we will need the raw output to have both logits and prediction masks to be able to compute the segmentation) the same way we handle the model for QA with beam search in XLNert. That class would then not be in the `AutoModelForSemanticSegmentation` as it doesn't have the same API, but we would need one version of `MaskFormerForSemanticSegmentation` with the same API (it might perform as well as this fancy one with some fine-tuning).